### ISSUE TITLE
Development environment | module "QtQuick" version 2.7 is not installed

### ISSUE THREAD

**Issue by topas-rec** [QTodoTxt/QTodoTxt2#56]
Hey everyone,

I was able to develop QTodoTxt in eclipse with Pydev plugin and installing some packages to fulfill dependencies.
Using this approach I cannot get QTodoTxt2 to run on my ubuntu 16.04. I had to install some qml and qtquick packages and fixed some error by doing so. When I run the app I get the following error that I cannot fix:

`QQmlApplicationEngine failed to load component`
`file:///[...]/git/QTodoTxt2/qtodotxt2/qml/QTodoTxt.qml:142 Type TaskListView unavailable`
`file:///[...]/git/QTodoTxt2/qtodotxt2/qml/TaskListView.qml:1 module "QtQuick" version 2.7 is not installed`

How do you develop the app? Which tools do you use and which packages / installations do you set up?

Best regards
topas

**Comment by oroulet** [comment]
I have 17.4 and simply run ./bin/qtodotxt
Make sure qt quick is installed

**Comment by topas-rec** [comment]
> I have 17.4 and simply run ./bin/qtodotxt Make sure qt quick is installed

Same result. Which dev path should I use?

**Comment by oroulet** [comment]
I have no dev path. I think you are missing the qt quick stuff . There is
another bug report from another user on that

**Comment by topas-rec** [comment]
I've noticed that report but mine is slightly different - so a new issue.

**Comment by topas-rec** [comment]
Can you tell me what you installed on the Ubuntu 17.4 to install qtquick?

**Comment by oroulet** [comment]
I have so many things installed I am not sure what id necessary or not but these might be necessary:
python3-pyqt5.qtquick
qml-module-qtquick2

**Comment by fuchsmich** [comment]
Try to change 2.7 to 2.4 in the qml file.

Am Samstag 24. Juni 2017 schrieb Topas:
> Can you tell me what you installed on the Ubuntu 17.4 to install qtquick?
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/QTodoTxt/QTodoTxt2/issues/56#issuecomment-31084189

-- 
Gesendet von meinem Jolla

**Comment by oroulet** [comment]
He has the same Ubuntu version as i have so that should work without
changes....

**Comment by topas-rec** [comment]
I hope I do not have these modules installed ;-) and will try to test your suggestions later.

I think it should work without changes. The dependencies should be clear.

**Comment by oroulet** [comment]
Let us know what you find. We should document this, in readme for example

**Comment by topas-rec** [comment]
> Try to change 2.7 to 2.4 in the qml file.

That tip above from @fuchsmich helps to resolve error `file:///[...]/git/QTodoTxt2/qtodotxt2/qml/TaskListView.qml:1 module "QtQuick" version 2.7 is not installed`

I saw some qml files import QtQuick 2.5 and some import QtQuick 2.7. Is that intended or should all qml files use the same QtQuick version?

Somehow these errors remain:
`QQmlApplicationEngine failed to load component`
`file:///media/topas/DATA/Office/Software development/git/QTodoTxt2/qtodotxt2/qml/QTodoTxt.qml:144 Type TaskListTableView unavailable`
`file:///media/topas/DATA/Office/Software development/git/QTodoTxt2/qtodotxt2/qml/TaskListTableView.qml:55 Type TaskLine unavailable`
`file:///media/topas/DATA/Office/Software development/git/QTodoTxt2/qtodotxt2/qml/TaskLine.qml:66 Cannot assign to non-existent property "onEditingFinished"`

**Comment by topas-rec** [comment]
`I have so many things installed I am not sure what id necessary or not but these might be necessary:
python3-pyqt5.qtquick
qml-module-qtquick2`

I've already got these packages installed.

**Comment by fuchsmich** [comment]
> I saw some qml files import QtQuick 2.5 and some import QtQuick 2.7. Is that intended or should all qml files use the same QtQuick version?

They rather should be the same. I have to clean up this mess. This comes from different PCs I'm working on with different versions of qt installed.
I think its a good idea to support qt 5.6. I have to check, which versions of qtquick modules belong to 5.6 and use them.

The other errors: Try a recent version.

**Comment by TRSx80** [comment]
> Let us know what you find. We should document this, in readme for example

I agree. I ran into similar issues when I installed recently. Unfortunately, I do not remember now exactly what I ended up installing either. :/ There are a lot of different Qt, QtQuick, etc. packages and it was very confusing to me as a first time user. And no documentation either. I just kept installing crap until it worked. lol But that's awful, we can and should do better.

Maybe I will reinstall from scratch, and keep better notes this time. But even that is only marginally useful, because I may have already installed some dependencies, previously, for other things... So perhaps one of you devs can figure out the dependencies and let us know? After that, I know it's a small thing but I would be happy to update wiki with some installation instructions or whatever.
